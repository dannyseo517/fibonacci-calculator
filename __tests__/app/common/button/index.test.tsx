import React from 'react';
import { shallow } from 'enzyme';
import Button from 'Common/button';

describe('Common button', () => {
    it('should render initial layout', () => {
        const callback = (): any => {
            return null;
        };
        const component = shallow(<Button buttonText='=' callback={callback}/>);
        expect(component.getElements()).toMatchSnapshot();
    });

    it('should call handleButtonClick on button click', () => {
        const callback = jest.fn();
        const component = shallow(<Button buttonText='=' callback={callback}/>);
        component.find('button').simulate('click');
        expect(callback.mock.calls.length).toEqual(1);
    });
});

