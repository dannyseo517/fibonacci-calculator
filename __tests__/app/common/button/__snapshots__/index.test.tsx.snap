// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common button should render initial layout 1`] = `
Array [
  <button
    onClick={[Function]}
  >
    =
  </button>,
]
`;
